### ISSUE TITLE
No assignment of previous assigned partitions after pod kill

### ISSUE THREAD

**Issue by SierraNL** [confluentinc/confluent-kafka-dotnet#1658]
Description
===========
We've seen this only once, and are not able to reproduce yet.
We have a solution with in total 40 partitions over 3 topics, shared over 3 pods in Kubernetes.
A single pod was killed by kubernetes for going over the memory limit (other issue).

But the issue was that after the restart, there was a rebalance, but the partitions the pod had assigned where not reassigned anymore. Not to the restarted pod, but also not to the other pods.
Since it was a kill, there was no option to gracefully shut down, and do an Unsubscribe.

It almost seemed like the pod was restarted within the heartbeat timeout so Kafka didn't notice.

My questions are:
- Are there timeout settings that need to be lower than the time a pod can restart to prevent this issue?
- Should you always call unsubscribe when starting up? To force the rebalance?
- Are there other ways to prevent this from happening?

Checklist
=========
 - Confluent.Kafka nuget version: 1.5.3
 - Apache Kafka version: 2.5.0
 - Client configuration.
 ```json
{
  "Kafka": {
    "Consumer": {
      "Config": {
        "bootstrap.servers": "myserver",
        "group.id": "fixed",
        "client.id": "fixed",
        "enable.auto.commit": "true",
        "auto.commit.interval.ms": "3000",
        "enable.auto.offset.store": "false",
        "auto.offset.reset": "earliest",
        "socket.nagle.disable": "true",
        "allow.auto.create.topics": "true"
      },
    },

    "Producer": {
      "Config": {
        "bootstrap.servers": "myserver",
        "client.id": "fixed",
        "retries": "5",
        "batch.num.messages": "1",
        "socket.nagle.disable": "true",
        "queue.buffering.max.ms": "0",
        "partitioner": "murmur2_random"
      }
    }
  }
}
```
 - [ ] Operating system: dotnet's default aspnet:3.1-buster-slim docker image

**Comment by mhowlett** [comment]
please try the latest version, and open a new issue if the problem persists. there has been a lot of work done to the consumer since 1.5.3.
